src/stores/VolunteerStore.js

```diff
@@ -28,23 +29,49 @@ export const useVolunteerStore = defineStore('volunteerStore', {
         })
         .finally(() => (this.fetching = false))
     },
+    async getVolunteer(volunteerId) {
+      //clear selected Volunteer
+      this.selectedVolunteer = null
+
+      // If there's no token, something went wrong
+      if (!this.token) throw Error('VoloDB-ERROR\n🙅‍♀️ ups! not logged in.')
+
+      this.fetching = true
+      await fetch(`${import.meta.env.VITE_BASE_URL}/volunteers/${volunteerId}`, {
+        method: 'GET',
+        headers: {
+          'content-type': 'application/json',
+          authorization: `Bearer ${this.token}`
+        }
+      })
+        .then((res) => {
+          if (!res.ok) throw Error(`VoloDB-ERROR\n🙅‍♀️ posting new volunteer failed! (${res.status}`)
+          return res.json()
+        })
+        .then((volunteer) => {
+          this.selectedVolunteer = volunteer
+        })
+        .finally(() => (this.fetching = false))
+    },
+
     async getVolunteers(pageNumber = 0) {
       this.fetching = true
+
+      // If there's no token, something went wrong
+      if (!this.token) throw Error('VoloDB-ERROR\n🙅‍♀️ ups! not logged in.')
+
       await fetch(`${import.meta.env.VITE_BASE_URL}/volunteers?page=${pageNumber}`, {
         method: 'GET',
         headers: {
           authorization: `Bearer ${this.token}`
         }
       })
         .then((res) => {
-          console.log(res.status)
-          if (!res.ok) {
-            throw Error(`ERROR:${res.status}`)
-          }
+          if (!res.ok) throw Error(`VoloDB-ERROR\n🙅‍♀️ posting new volunteer failed! (${res.status}`)
```

Comment:
needs another error message: "fetching volunteers failed.." or something like this?